Code/Framework/AzNetworking/AzNetworking/DataStructures/TimeoutQueue.h

```diff
@@ -86,6 +86,10 @@ namespace AzNetworking
         TimeoutId        m_nextTimeoutId = TimeoutId{ 0 };
         TimeoutItemMap   m_timeoutItemMap;
         TimeoutItemQueue m_timeoutItemQueue;
+
+        // TimeoutQueue is a shared resource among connections. A mutex (or a read-write sync object) is required with multi-threaded sends.
+        // See @sv_multithreadedConnectionUpdates cvar.
+        AZStd::mutex m_mutex;
```

Comment:
This works for now but it feels like a lockless scheme could be devised here. Maybe a bucket per thread or similar division to eliminate contention.

Reply:
Agreed, definitely revisit in the future.